webapp/src/Controller/Jury/ShadowDifferencesController.php

```diff
@@ -0,0 +1,207 @@
+<?php declare(strict_types=1);
+
+namespace App\Controller\Jury;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query\Expr\Join;
+use App\Controller\BaseController;
+use App\Entity\ExternalJudgement;
+use App\Entity\Judging;
+use App\Entity\Submission;
+use App\Service\DOMJudgeService;
+use App\Service\SubmissionService;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/jury/shadow-differences")
+ * @IsGranted("ROLE_ADMIN")
+ */
+class ShadowDifferencesController extends BaseController
+{
+    /**
+     * @var DOMJudgeService
+     */
+    protected $dj;
+
+    /**
+     * @var SubmissionService
+     */
+    protected $submissions;
+
+    /**
+     * @var SessionInterface
+     */
+    protected $session;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    protected $em;
+
+    public function __construct(
+        DOMJudgeService $dj,
+        SubmissionService $submissions,
+        SessionInterface $session,
+        EntityManagerInterface $em
+    ) {
+        $this->dj          = $dj;
+        $this->submissions = $submissions;
+        $this->session     = $session;
+        $this->em          = $em;
+    }
+
+    /**
+     * @Route("", name="jury_shadow_differences")
+     */
+    public function indexAction(Request $request)
+    {
+        $shadowMode = DOMJudgeService::DATA_SOURCE_CONFIGURATION_AND_LIVE_EXTERNAL;
+        $dataSource = $this->dj->dbconfig_get('data_source', DOMJudgeService::DATA_SOURCE_LOCAL);
+        if ($dataSource != $shadowMode) {
+            $this->addFlash('danger', sprintf(
+                'Shadow differences only supported when data_source is %d',
```

Comment:
This is not an actionable comment: in the long run, I would like to even import results from the shadow when we're primary.

Reply:
What do you suggest, that we remove this limitation in full?

Reply:
As I said: my comment is not actionable. I would not do anything right now but keep in mind that even in primary mode we might want to show diffs to shadow on our side (and don't add anything that makes this harder).